src/test/java/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDaoTest.java

```diff
@@ -675,6 +676,92 @@ public void constraintRulesOnQuestionOverwriteConstraintRules() {
         assertTrue(rule.getEndSurvey());
     }
     
+    @Test
+    public void getSurveyAllVersionsIncludeDeletedByIdentifier() {
+        List<DynamoSurvey> surveyList = ImmutableList.of(new DynamoSurvey(), new DynamoSurvey());
+        mockSurveyMapper(surveyList);
+        
+        
+        List<Survey> results = dao.getSurveyAllVersions(TEST_STUDY, SURVEY_IDENTIFIER, true);
+        assertEquals(results, surveyList);
+        
+        verify(mockSurveyMapper).queryPage(eq(DynamoSurvey.class), queryCaptor.capture());
+        DynamoDBQueryExpression<DynamoSurvey> query = queryCaptor.getValue();
+        assertEquals(query.getHashKeyValues().getIdentifier(), SURVEY_IDENTIFIER.substring(11));
+        
+        Condition rangeKeyCondition = query.getRangeKeyConditions().get("identifier");
+        assertEquals(rangeKeyCondition.getComparisonOperator(), EQ.name());
+        assertEquals(rangeKeyCondition.getAttributeValueList().get(0).getS(), SURVEY_ID);
+        
+        assertNull(query.getQueryFilter());
+    }
+    
+    @Test
+    public void getSurveyAllVersionsExcludeDeletedByIdentifier() {
+        List<DynamoSurvey> surveyList = ImmutableList.of(new DynamoSurvey(), new DynamoSurvey());
+        mockSurveyMapper(surveyList);
+        
+        List<Survey> results = dao.getSurveyAllVersions(TEST_STUDY, SURVEY_IDENTIFIER, false);
+        assertEquals(results, surveyList);
+        
+        verify(mockSurveyMapper).queryPage(eq(DynamoSurvey.class), queryCaptor.capture());
+        DynamoDBQueryExpression<DynamoSurvey> query = queryCaptor.getValue();
+        
+        Condition rangeKeyCondition = query.getRangeKeyConditions().get("identifier");
+        assertEquals(rangeKeyCondition.getComparisonOperator(), EQ.name());
+        assertEquals(rangeKeyCondition.getAttributeValueList().get(0).getS(), SURVEY_ID);
+        
+        assertEquals(query.getQueryFilter().size(), 1);
+        verifyIsDeletedQueryCondition(query);
+    }
+    
+    @Test
+    public void getSurveyMostRecentVersionByIdentifier() {
+        DynamoSurvey persisted = new DynamoSurvey();
+        mockSurveyMapper(persisted);
```

Comment:
Consider creating multiple surveys with createdOn in some arbitrary order to make sure that we're sorting it correctly and returning the most recent one. (Same with getSurveyMostRecentlyPublishedVersion().)